src/neovim_api.rs

```diff
@@ -1009,13 +1012,17 @@ where
       .map_err(|v| Box::new(CallError::WrongValueType(v)))
   }
 
-  pub async fn call_function(
+  pub async fn call_function<T: Serialize + fmt::Debug>(
     &self,
     fname: &str,
-    args: Vec<Value>,
+    args: T,
   ) -> Result<Value, Box<CallError>> {
+
+    #[derive(Debug, Serialize)]
+    struct Args<'a, T: Serialize>(&'a str, T);
+
     self
-      .call("nvim_call_function", call_args![fname, args])
+      .call("nvim_call_function", Args(fname, args))
```

Comment:
Are there any blanket impl's we could take advantage of here? Maybe we can just use `(fname, args)`? Otoh, this might just be clearer to read.